app-backend/backsplash-server/src/main/scala/com/rasterfoundry/backsplash/middleware/AccessLoggingMiddleware.scala

```diff
@@ -0,0 +1,56 @@
+package com.rasterfoundry.backsplash.middleware
+
+import org.http4s.HttpRoutes
+import org.http4s.Request
+import cats.data.Kleisli
+import io.circe.syntax._
+import org.http4s.util.CaseInsensitiveString
+import cats.effect.Sync
+import com.typesafe.scalalogging.Logger
+import java.time.Instant
+
+class AccessLoggingMiddleware[F[_]: Sync](
+    service: HttpRoutes[F],
+    logger: Logger
+) {
+  def withLogging(enabled: Boolean) = {
+    if (!enabled) service
+    else {
+      Kleisli { (request: Request[F]) =>
+        val requestStart = Instant.now
+        val headerWhitelist: Set[CaseInsensitiveString] =
```

Comment:
I'm not logging every header for two reasons --

I don't know in advance a list of sensitive headers to exclude, and I think we've had issues in the past with large logging payloads causing performance degradation. As a defensive measure, I'm keeping things that I know are useful and not sensitive and leaving out anything else.